src/value.c

```diff
@@ -670,15 +670,11 @@ int RSValue_SendReply(RedisModule_Reply *reply, const RSValue *v, SendReplyFlags
       return RedisModule_Reply_String(reply, v->rstrval);
 
     case RSValue_Number: {
-      if (!(flags & SENDREPLY_FLAG_EXPAND)) {
+      if (!(flags & SENDREPLY_FLAG_EXPAND) && !(flags & SENDREPLY_FLAG_TYPED)) {
+        // With STRING format and with `_NUM_SSTRING` disabled, return the number as a string
         char buf[128];
         size_t len = RSValue_NumToString(v->numval, buf);
-
-        if (flags & SENDREPLY_FLAG_TYPED) {
-          return RedisModule_Reply_Error(reply, buf);
-        } else {
-          return RedisModule_Reply_StringBuffer(reply, buf, len);
-        }
+        return RedisModule_Reply_StringBuffer(reply, buf, len);
```

Comment:
Why not use existing PR https://github.com/RediSearch/RediSearch/pull/5187

Reply:
OK. I'll include this changes in the existing PR.